lumen/target.py

```diff
@@ -84,6 +84,10 @@ def _construct_layout(self):
     def from_spec(cls, spec, filters=None, pipelines={}):
         spec = dict(spec)
         view_specs = spec.get('views', [])
+        if view_specs is None:
+            # If views is not defined in yaml file
+            raise ValueError("No 'views' was provided during instantiation.")
```

Comment:
This branch will never be declared because the get returns an empty list:

```python
view_specs = spec.get('views', [])
```

Reply:
Got it with this yaml file
``` yaml
config:
  title: Lumen

sources:
  penguins:
    type: file
    tables:
      penguins: https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-07-28/penguins.csv

targets:
  - title: Table
    source: penguins
    view:  # <- Missing an s
      table:
        type: table
        table: penguins
```

Reply:
Oh right, this is on `Card` not on `target`. I'll move the validation somewhere earlier than this. The goal is also to add very specific error messages that can help users find where in the YAML specification the error is. So the goal here would be to have some like this be the output:

```
Ensure the following target declares a `views` field:

    - title: Table
       source: penguins
       view: # <- Missing an s
         table:
           type: table
           table: penguins
```

Mapping it back to the YAML is obviously quite difficult but would be nice. In the absence of that it should at least mention the title for instance.